tests/composite_pk/test_values.py

```diff
@@ -0,0 +1,216 @@
+                    (self.user_3.pk, self.user_3.id),
+                ),
+            )
+
+    def test_values(self):
+        with self.subTest('User.objects.values("pk")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk").order_by("pk"),
+                (
+                    {"pk": self.user_1.pk},
+                    {"pk": self.user_2.pk},
+                    {"pk": self.user_3.pk},
+                ),
+            )
+        with self.subTest('User.objects.values("pk", "email")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk", "email").order_by("pk"),
+                (
+                    {"pk": self.user_1.pk, "email": self.USER_1_EMAIL},
+                    {"pk": self.user_2.pk, "email": self.USER_2_EMAIL},
+                    {"pk": self.user_3.pk, "email": self.USER_3_EMAIL},
+                ),
+            )
+        with self.subTest('User.objects.values("pk", "id")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk", "id").order_by("pk"),
+                (
+                    {"pk": self.user_1.pk, "id": self.user_1.id},
+                    {"pk": self.user_2.pk, "id": self.user_2.id},
+                    {"pk": self.user_3.pk, "id": self.user_3.id},
+                ),
+            )
+        with self.subTest('User.objects.values("pk", "tenant_id", "id")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk", "tenant_id", "id").order_by("pk"),
+                (
+                    {
+                        "pk": self.user_1.pk,
+                        "tenant_id": self.user_1.tenant_id,
+                        "id": self.user_1.id,
+                    },
+                    {
+                        "pk": self.user_2.pk,
+                        "tenant_id": self.user_2.tenant_id,
+                        "id": self.user_2.id,
+                    },
+                    {
+                        "pk": self.user_3.pk,
+                        "tenant_id": self.user_3.tenant_id,
+                        "id": self.user_3.id,
+                    },
+                ),
+            )
+        with self.subTest('User.objects.values("pk", "pk")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk", "pk").order_by("pk"),
+                (
+                    {"pk": self.user_1.pk},
+                    {"pk": self.user_2.pk},
+                    {"pk": self.user_3.pk},
+                ),
+            )
+        with self.subTest('User.objects.values("pk", "id", "pk", "id")'):
+            self.assertSequenceEqual(
+                User.objects.values("pk", "id", "pk", "id").order_by("pk"),
+                (
+                    {"pk": self.user_1.pk, "id": self.user_1.id},
+                    {"pk": self.user_2.pk, "id": self.user_2.id},
+                    {"pk": self.user_3.pk, "id": self.user_3.id},
+                ),
+            )
```

Comment:
One thing that is dearly missing here is any form of aggregation over an explicit `values` grouping

e.g.

```python
User.objects.values("pk").annotate(Count("comments"))
```

Which should generate a query of the form

```sql
SELECT user.tenant_id, user.id, COUNT(comment.id)
FROM user
LEFT JOIN comment ON (comment.tenant_id = user.tenant_id AND comment.user_id = user.id)
GROUP BY 1, 2
```

I suspect a similar problem might arise when [implicit grouping by `pk` due to `features.allows_group_by_selected_pks`](https://github.com/django/django/blob/f16a9a556fb4225f9094048614f4fcec3db7e067/django/db/models/sql/compiler.py#L201-L226).

```python
User.objects.annotate(Count("comments"))
```

Which should generate a query of the following form when `allows_group_by_selected_pks = True`

```sql
SELECT user.tenant_id, user.id, user.*, COUNT(comment.id)
FROM user
LEFT JOIN comment ON (comment.tenant_id = user.tenant_id AND comment.user_id = user.id)
GROUP BY 1, 2
```

It might already be working but we should have tests exercising it if that's the case.

Reply:
I've documented this here: https://github.com/django/django/pull/18056/files#diff-cca8870fcaec19104d999f61553ba925c72e2eb19b4933068c4849f2ce58a6f6R151

`Count("comments")` is an unfortunate edge case, but users can use `Count("comments__id")` instead.

Reply:
I pushed a test (`test_get_user_values_annotated_with_comments_id_count`) for `User.objects.values("pk").annotate(Count("comments"))`.

Reply:
Please check the `CompositePKAggregateTests` tests.

Reply:
> Count("comments") is an unfortunate edge case, but users can use Count("comments__id") instead.

Oh yeah I remember now, it's due to the arity of `COUNT`. I remember we discussed this exact issue a few weeks ago now 🙈 .

That's great, my concern were more around making sure the proper `GROUP BY` was generated over `Count("comments__id")` over `Count("comments")`.

Reply:
I added support for `Count("comments")` based on your suggestions.